File: Algorithms/src/1411. Number of Ways to Paint N × 3 Grid/Code/solution.go
func numOfWays(n int) int {
    const MOD int = 1e9 + 7

    same := 6
    diff := 6

    for i := 2; i <= n; i++ {
        newSame := (same*3 + diff*2) % MOD
        newDiff := (same*2 + diff*2) % MOD

        same = newSame
        diff = newDiff
    }

    return (same + diff) % MOD
}

File: Algorithms/src/1411. Number of Ways to Paint N × 3 Grid/Code/solution.js
var numOfWays = function (n) {
  const MOD = 1e9 + 7;

  let same = 6;
  let diff = 6;

  for (let i = 2; i <= n; i++) {
    const newSame = (same * 3 + diff * 2) % MOD;
    const newDiff = (same * 2 + diff * 2) % MOD;

    same = newSame;
    diff = newDiff;
  }

  return (same + diff) % MOD;
};
